src/System.Linq/src/System/Linq/Skip.cs

```diff
@@ -121,5 +121,33 @@ private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSour
                 }
             }
         }
+
+#if netcoreapp11
+        public static IEnumerable<TSource> SkipLast<TSource>(this IEnumerable<TSource> source, int count)
+        {
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+
+            return SkipLastIterator(source, Math.Max(count, 0));
```

Comment:
If we had `if (count <= 0) return Skip (source, 0)` then:
1. We get it's check for the source to such a call being a linq-internal object in which case it was considered reasonably safe to just return the source.
2. Otherwise we get a lighter implementation of the pass-through, without allocating a queue and then repeatedly enqueuing and dequeuing it.